Sources/MessagingPush/PushHandling/PushEventHandlerProxy.swift

```diff
@@ -81,7 +84,10 @@ class PushEventHandlerProxyImpl: PushEventHandlerProxy {
             return
         }
 
-        Task {
+        // UserNotification runs this event on the main thread.
+        // Run this async task on the main thread to match that behavior. Otherwise, we run the risk of warnings or crashes by trying to call UIKit
+        // functions from a background thread.
+        Task { @MainActor in
```

Comment:
I can't recall all the syntax requirements, but if we don't need to wrap this into `Task` then let's remove it.

Reply:
Because the `shouldDisplayPushAppInForeground` function is not `async`, creating a `Task` is required.